docs/concepts/libp2p.md

```diff
@@ -9,68 +9,125 @@ related:
+data to them, and maintain the network topology. In IPFS, discovery and routing occur concurrently, making it a dynamic and resilient network that can adapt to changes in network conditions.
+
+<!--Check out the libp2p documentation to learn more about peer discovery and routing
+in libp2p. -->
+
+### Content routing
+
+Content routing refers to the process of identifying a specific peer that holds certain data and the means to connect to them within a P2P network.
+
+IPFS utilizes content routers to identify peers that have requested data and to inform the network that a peer can provide
+a certain piece of content. In IPFS, content routing is achieved through a Distributed Hash Table (DHT), specifically Kademlia. Each piece of content is assigned a unique identifier, and the DHT stores the information of the closest peer holding the
+content. This allows for efficient routing and reduces the number of possible routes to find a specific piece of data.
+
+<!--Check out the libp2p documentation to learn more about content routing
+in libp2p. -->
+
+### NAT traversal
+
+Network Address Translation (NAT) allows peers to move traffic between network boundaries. NAT maps an address from one address space to another. While NAT is usually transparent for outgoing connections, listening for incoming connections requires some configuration. libp2p has the following main approaches to NAT traversal available:
+
+- [Automatic router configuration](https://docs.libp2p.io/concepts/nat/overview/#automatic-router-configuration) for port forwarding.
+
+> - [UPnP (Universal Plug and Play)](https://en.wikipedia.org/wiki/Universal_Plug_and_Play)
+> - [NAT-PMP (NAT Port Mapping Protocol)](https://en.wikipedia.org/wiki/NAT_Port_Mapping_Protocol)
+
+Manually opening a port requires technical expertise and does not enforce authentication
+or authorization. In many settings, UPnP is disabled by the router or a firewall. UPnP may also not work depending on the
+router’s firmware.
+
+As a result, libp2p introduced a [hole punching technique](https://docs.libp2p.io/concepts/nat/hole-punching/) that makes use of the following protocols:
+
+- [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/) allows a node to request other peers to dial its presumed public addresses. This is equivalent to the [STUN protocol](https://www.rfc-editor.org/rfc/rfc3489) in ICE.
+- [Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/) establishes and routes a secure relay connection through a public relay node. This is equivalent to the [TURN protocol](https://datatracker.ietf.org/doc/html/rfc5766) in ICE.
+- DCUtR (Direct Connection Upgrade through Relay) establishes a direct connection between nodes through hole punching without a signaling server.
+
+Check out the [libp2p documentation](https://docs.libp2p.io/concepts/nat/overview/) to learn more about NATs in libp2p.
```

Comment:
```suggestion
Libp2p provides [automatic router configuration](https://docs.libp2p.io/concepts/nat/overview/#automatic-router-configuration) for port forwarding.

- [UPnP (Universal Plug and Play)](https://en.wikipedia.org/wiki/Universal_Plug_and_Play)
- [NAT-PMP (NAT Port Mapping Protocol)](https://en.wikipedia.org/wiki/NAT_Port_Mapping_Protocol)

Manually opening a port requires technical expertise, and does not enforce authentication
or authorization. In many settings, UPnP is disabled by the router or a firewall. UPnP may also not work depending on the router’s firmware.

Because of this, libp2p introduced a [hole punching technique](https://docs.libp2p.io/concepts/nat/hole-punching/) that makes use of the following protocols:

- [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/), which allows a node to request other peers to dial its presumed public addresses. This is equivalent to the [STUN protocol](https://www.rfc-editor.org/rfc/rfc3489) in ICE.
- [Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/), which establishes and routes a secure relay connection through a public relay node. This is equivalent to the [TURN protocol](https://datatracker.ietf.org/doc/html/rfc5766) in ICE.
- Direct Connection Upgrade through Relay (DCUtR) establishes a direct connection between nodes through hole punching without the use of a signaling server.

For more information about NATs in libp2p., see the [libp2p documentation](https://docs.libp2p.io/concepts/nat/overview/).
```